### PR TITLE
Fix addConnection for bus connections

### PR DESCRIPTION
**Pull request by lochel**
### Purpose

* Remove unneccessary restrictions to bus connections
* Test if connection exists already, before adding a new connection
* Fix getConnection by making it commutative.

### Type of Change

- Bug fix (non-breaking change which fixes an issue)

### Checklist

- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by OpenModelica-Hudson** [comment]
The following pull requests are now being [tested](https://test.openmodelica.org/hudson/job/OMSimulator-PR/751//console). It may take a while, depending on the workload.
  - OpenModelica/OMSimulator#443
  - OpenModelica/OMSimulator-testsuite#104

The changes will be pushed if all tests run successfully.

**Comment by OpenModelica-Hudson** [comment]
All tests were successful. :blush:

File: src/OMSimulatorLib/System.cpp [hunk 1]
```diff
@@ -1092,7 +1082,7 @@ oms_status_enu_t oms3::System::setTLMBusGeometry(const oms3::ComRef& cref, const
 oms3::Connection* oms3::System::getConnection(const oms3::ComRef& crefA, const oms3::ComRef& crefB)
 {
   for (auto& connection : connections)
-    if (connection && connection->getSignalA() == crefA && connection->getSignalB() == crefB)
+    if (connection && (connection->getSignalA() == crefA && connection->getSignalB() == crefB || connection->getSignalA() == crefB && connection->getSignalA() == crefB))
```
Comment (adeas31) [inline src/OMSimulatorLib/System.cpp]: This looks very weird. I think you probably want this,

`if (connection && ((connection->getSignalA() == crefA && connection->getSignalB() == crefB) || (connection->getSignalA() == crefB && connection->getSignalB() == crefA)))`

Comment (lochel) [reply src/OMSimulatorLib/System.cpp]: Thanks, I will fix it. 👍

**Pull request closed without merging**